js/src/setup-mc/setup-stepper/setup-paid-ads/clientSession.js

```diff
@@ -0,0 +1,59 @@
+/**
+ * @typedef { import(".~/data/actions").CountryCode } CountryCode
+ *
+ * @typedef {Object} CampaignData
+ * @property {number|undefined} amount Daily average cost of the paid ads campaign.
+ * @property {Array<CountryCode>} countryCodes Audience country codes of the paid ads campaign. Example: 'US'.
+ */
+
+const KEY_SHOW_PAID_ADS_SETUP = 'gla-onboarding-show-paid-ads-setup';
+const KEY_PAID_ADS = 'gla-onboarding-paid-ads';
+
+const { sessionStorage } = window;
+
+const clientSession = {
+	/**
+	 * @param {boolean} isShowing Whether the paid ads setup is showing.
+	 */
+	setShowPaidAdsSetup( isShowing ) {
+		const showing = Boolean( isShowing ).toString();
+		sessionStorage.setItem( KEY_SHOW_PAID_ADS_SETUP, showing );
+	},
+
+	/**
+	 * @param {boolean} defaultValue The default value to be returned if stored value is not available.
+	 * @return {boolean} Returns the stored value. It will return `defaultValue` if stored value is not available.
+	 */
+	getShowPaidAdsSetup( defaultValue ) {
+		const showing = sessionStorage.getItem( KEY_SHOW_PAID_ADS_SETUP );
+		return showing === null ? defaultValue : showing === 'true';
+	},
+
+	/**
+	 * @param {CampaignData} data Campaign data to be stored.
+	 * @param {number|undefined} data.amount Daily average cost of the campaign.
+	 * @param {Array<CountryCode>} data.countryCodes Country codes of the campaign.
+	 */
+	setCampaign( { amount, countryCodes } ) {
+		try {
+			const json = JSON.stringify( { amount, countryCodes } );
+			sessionStorage.setItem( KEY_PAID_ADS, json );
+		} catch ( e ) {
+			// noop
+		}
```

Comment:
> Why do we need this `try...catch` here?

To tolerate unexpected/unknown errors, I sometimes add a `try catch` when converting JSON.

> What kind of errors do we want to silence?

The unexpected/unknown error can not be recovered but the failure doesn't make the paid ads setup unworkable, so I left it to silence.

---

Considering this isn't common util for the whole codebase but only for the paid ads setup, it should be fine without the `try catch`. Removed the `try catch` in https://github.com/woocommerce/google-listings-and-ads/pull/1683/commits/239b45f78e36fa23addf1294b0bdc98c608a12f0

Reply:
:) Thanks for the explanations. I think as you mentioned this is local util, that's us to provide `amount, countryCodes` and we have no actual means to fall back from such error. I'd rather not silence the error and fails fast and explicitly, so it would be easier to identify and report for HEs